drivers/Ticker.h

```diff
@@ -78,9 +78,9 @@ class Ticker : public TimerEvent, private NonCopyable<Ticker> {
      *  @param func pointer to the function to be called
      *  @param t the time between calls in seconds
      */
-    void attach(Callback<void()> func, float t)
+    void attach(Callback<void()> func, const s_timestamp_t t)
```

Comment:
Indeed, this changes the behaviour of existing code. `attach(0.5)` will be treated as `attach(0)`. 

You should get a warning, but that could be missed.

Reply:
c86ac7e

Reply:
Unfortunately, this override won't really work, unless I'm mistaken.

I believe `attach(2)` will be ambiguous - `2` is neither a `uint64_t` nor a `float`, and it won't know which overload to prefer.

We could add overrides for `std::duration`, as I suggested below, but not for more implicitly-convertable arithmetic types :(

To deprecate, it would have to be an actual new API like `attach_s`.

Reply:
The change I made make the user more conscious of what s/he is doing. It will have to deliberately either cast 2 to `s_timestamp_t` or define a variable of type `s_timestamp_t`. This is a good thing. Passing just 2 causes a compilation error.

Reply:
Granted, that's better than a silent behaviour change, but making existing code not compile is a lot stronger than a deprecation, so I'm not sure whether it's merited.

(But it's also not strongly typed either - they could pass a `us_timestamp_t`, and it wouldn't bat an eyelid.)

Reply:
attach_s is a good suggestion since we already have attach_us.

Reply:
Ok I went for `Ticker::attach_s()`, 1709847

Reply:
That's great.